### ISSUE TITLE
Problem in installing dependencies 

### ISSUE THREAD

**Issue by elahe-badali** [alexzhou907/PVD#5]
Hi, thank you for sharing your code. I have some problems in running the code. My system is Windows 10. 
![image](https://user-images.githubusercontent.com/39738835/167309617-79cb7d70-5e86-40d0-aa0e-a532b51fafa4.png)

**Comment by albertotono** [comment]
did you compile also the other extensions? like CD? and check your gcc to be 8 like in PVCNN. I am on Linux, not sure if this helps.

**Comment by elahe-badali** [comment]
I finally decided to run it on Linux and everything is ok.

**Comment by FengLoveBella** [comment]
@albertotono @blue-sky54 I was wondering did you run this code? I encounter the question of "No module named '_pvcnn_backend'  when I try to run this code. My system is Ubuntu18.04, Cuda is 10.1, Pytorch is 1.4.0. Also, I tried Ubuntu 18.04 cuda 10.2, pytorch 1.7 still have this problem.

**Comment by FengLoveBella** [comment]
@albertotono @blue-sky54 
![Screenshot from 2022-12-16 16-02-56](https://user-images.githubusercontent.com/11606333/208051954-bbf72122-0915-4feb-9900-33e0cc409f11.png)

**Comment by albertotono** [comment]
Yes, we run, we compared against for 3D-LDM https://arxiv.org/abs/2212.00842 and other projects. 

PVCNN (https://pvcnn.mit.edu/) is used as the backbone of the model, check the modules, but it looks also issue with the jit compilation I think. Did you compile all the CD, EMD? and other?

**Comment by FengLoveBella** [comment]
@albertotono Thanks for the reply. I compiled all the CD and EMD, however, I still encountered the "No module named _pvcnn_backend" problem. I tried to change the PyTorch (from 1.5.0 to 1.7.0 to 1.4.0) and Cuda versions (from 10.2 to 10.1 to 11.2), but every version has the same problem as mentioned above.

**Comment by albertotono** [comment]
Am I wrong or it should be an issue with the cpp_extension for torch in this case . maybe this could help https://pytorch.org/tutorials/advanced/cpp_extension.html

**Comment by elahe-badali** [comment]
@FengLoveBella 
Hi there, yes after :
```!python setup.py install -q  
cp build/**/emd_cuda.cpython-37m-x86_64-linux-gnu.so .
```
I run these two commands:
```
!apt install cuda gcc-8 g++-8 -y -qq;
!apt install gcc-8 g++-8 -y
```

**Comment by FengLoveBella** [comment]
@albertotono Thanks for the link, I already read this page, however, this problem isn't solved. @blue-sky54 I did this step "cp build/**/emd_cuda.cpython-37m-x86_64-linux-gnu.so ." and the *.so file is already in the same folder with EMD. @blue-sky54 gcc-8 and g++-8 have already been installed in the system as shown in the following figure.
![image](https://user-images.githubusercontent.com/11606333/208228857-0f8c555f-4b29-4403-9705-f480482327eb.png)

**Comment by angelia68** [comment]
> @albertotono感谢您的链接，我已经阅读了此页面，但是，这个问题没有解决。@blue-sky54 我做了这一步“cp build/**/emd_cuda.cpython-37m-x86_64-linux-gnu.so”。*.so 文件已与 EMD 位于同一文件夹中。@blue-sky54 gcc-8和g++-8已经安装在系统中如下图所示。 ![图片](https://user-images.githubusercontent.com/11606333/208228857-0f8c555f-4b29-4403-9705-f480482327eb.png)

Have you solved it? I also encountered this problem.

**Comment by goballooning** [comment]
> Am I wrong or it should be an issue with the cpp_extension for torch in this case . maybe this could help https://pytorch.org/tutorials/advanced/cpp_extension.html

Hi, would you mind sharing your setup? 
I'm using an Arch Linux host and GTX 3090 with latest driver. However, I cannot compile PyTorchEMD in a Docker container: Ubuntu 18.04 with PyTorch1.4, Cuda 10.1 as specified on https://github.com/alexzhou907/PVD. Should I find another Nvidia card that supports Cuda 10.1?

I just replaced the 3090 with a 1050 Ti and it compiled properly.

**Comment by Lzars** [comment]
> did you compile also the other extensions? like CD? and check your gcc to be 8 like in PVCNN. I am on Linux, not sure if this helps.

My cuda=11.3, pytorch=1.12.0. When running train_completion.py, it got stuck in load() in backend.py. What is the reason?
